src/main/java/org/dita/dost/module/reader/TopicReaderModule.java

```diff
@@ -180,54 +184,43 @@ public void readStartFile() throws DITAOTException {
         final List<Reference> res = new ArrayList<>();
         assert startFileInfo.src != null;
         final URI tmp = job.tempDirURI.resolve(startFileInfo.uri);
-        final Source source = job.getStore().getSource(tmp);
-        logger.info("Reading " + tmp);
         try {
-            final XMLStreamReader in = XMLInputFactory.newInstance().createXMLStreamReader(source);
-            while (in.hasNext()) {
-                int eventType = in.next();
-                switch (eventType) {
-                    case START_ELEMENT:
-                        final String cls = in.getAttributeValue(null, ATTRIBUTE_NAME_CLASS);
-                        if (!MAP_TOPICREF.matches(cls)) {
-                            break;
-                        }
-                        final URI href = getHref(in);
-                        if (href != null) {
-                            FileInfo fi = job.getFileInfo(startFileInfo.src.resolve(href));
-                            if (fi == null) {
-                                fi = job.getFileInfo(tmp.resolve(href));
-                            }
-                            assert fi != null;
-                            assert fi.src != null;
-                            String format = in.getAttributeValue(DITA_OT_NS, ATTRIBUTE_NAME_ORIG_FORMAT);
-                            if (format == null) {
-                                format = in.getAttributeValue(null, ATTRIBUTE_NAME_FORMAT);
-                            }
-                            res.add(new Reference(fi.src, format));
-                            nonConrefCopytoTargetSet.add(fi.src);
-                        }
-                        break;
-                    default:
-                        break;
+            final XdmNode source = job.getStore().getImmutableNode(tmp);
+            logger.info("Reading " + tmp);
+            final Predicate<? super XdmNode> isTopicref = xdmItem -> MAP_TOPICREF.matches(xdmItem.getAttributeValue(QNAME_CLASS));
```

Comment:
Off-topic, In cases like this, local variable in Java 10 would be nice for readability:

```java
var isTopicref = xdmItem -> MAP_TOPICREF.matches(xdmItem.getAttributeValue(QNAME_CLASS));
```